concreate/descriptor.py

```diff
@@ -195,13 +195,60 @@ def __init__(self, descriptor):
         map:
           repositories:
             seq:
-              - {type: str}
+              - {type: any}
           install:
             seq:
-            - {type: str}""")]
+              - {type: str}"""),
+                        yaml.safe_load("""
+        map:
+          repositories:
+            seq:
+              - {type: any}
+          packages:
+            seq:
+              - {type: any}""")]
 
         super(Packages, self).__init__(descriptor)
 
+        if 'install' in self.descriptor:
+            self.descriptor['packages'] = []
+            for pkg in self.descriptor['install']:
+                self.descriptor['packages'].append(ListItem(pkg))
+        elif 'packages' in self.descriptor:
+            pkgs = self.descriptor['packages'][:]
+            self.descriptor['packages'] = []
+            for pkg in pkgs:
+                self.descriptor['packages'].append(ListItem(pkg))
+
+        if 'repositories' in self.descriptor:
+            repos = self.descriptor['repositories'][:]
+            self.descriptor['repositories'] = []
+            for repo in repos:
+                self.descriptor['repositories'].append(ListItem(repo))
```

Comment:
Above code looks like some repetition, but I have no idea how to write it better/clearer.